arangod/RestHandler/RestSchemaHandler.cpp

```diff
@@ -0,0 +1,721 @@
+    return propRes;
+  }
+  dataBuilder.close();
+
+  velocypack::Builder viewBuilder;
+  viewBuilder.add(velocypack::Value(velocypack::ValueType::Object));
+  viewBuilder.add("viewName", velocypack::Value(viewName));
+  viewBuilder.add("links", velocypack::Value(velocypack::ValueType::Array));
+  auto data = dataBuilder.slice();
+  if (data.hasKey("links")) {  // If not, linksBuilder will be an empty array
+    for (auto li : velocypack::ObjectIterator(data.get("links"))) {
+      auto colName = li.key.copyString();
+      auto colValue = li.value;
+      TRI_ASSERT(colValue.isObject() && colValue.hasKey("fields") &&
+                 colValue.hasKey("includeAllFields") &&
+                 colValue.hasKey("analyzers"));
+      viewBuilder.add(velocypack::Value(velocypack::ValueType::Object));
+      viewBuilder.add("collectionName", velocypack::Value(colName));
+      viewBuilder.add("fields", velocypack::ValueType::Array);
+      std::set<std::string> includedAttrSet;
+      for (auto fi : velocypack::ObjectIterator(colValue.get("fields"))) {
+        viewBuilder.add(velocypack::Value(velocypack::ValueType::Object));
+        TRI_ASSERT(fi.value.hasKey("analyzers"));
+
+        viewBuilder.add("attribute", velocypack::Value(fi.key.copyString()));
+        viewBuilder.add("analyzers", fi.value.get("analyzers"));
+        viewBuilder
+            .close();  // Closing object -> {attribute: ***, analyzers: ***}
+        includedAttrSet.insert(fi.key.copyString());
+      }
+      viewBuilder.close();  // Closing array -> fields: [{}, {}]
+      if (colValue.get("includeAllFields").isTrue()) {
+        viewBuilder.add("allAttributeAnalyzers", colValue.get("analyzers"));
+      }
+      viewBuilder
+          .close();  // Closing object -> {collectionName: ***, fields: []}
+      colSet.insert(colName);
+    }
+  }
+  viewBuilder.close();  // Closing array -> links: [{}, {}]
+  viewBuilder.close();  // Closing object -> {viewName: ***, links: []}
+
+  viewsArrBuilder.add(viewBuilder.slice());
+  return Result{};
+}
+
+Result RestSchemaHandler::getAllViewsAndCollections(
+    velocypack::Builder& viewsBuilder, std::set<std::string>& colSet) {
+  std::vector<LogicalView::ptr> views;
+  LogicalView::enumerate(_vocbase,
+                         [&views](LogicalView::ptr const& view) -> bool {
+                           views.emplace_back(view);
+                           return true;
+                         });
+
+  velocypack::Builder viewsArrBuilder;
+  viewsArrBuilder.openArray();
+  for (auto view : views) {
+    TRI_ASSERT(view != nullptr);
+    if (!view) {
+      continue;
+    }
+    Result viewRes =
+        getViewAndCollections(view->name(), viewsArrBuilder, colSet);
+    if (viewRes.fail()) {
+      return viewRes;
+    }
+  }
+  viewsArrBuilder.close();
+  viewsBuilder.add("views", viewsArrBuilder.slice());
+  return Result{};
+}
+
+Result RestSchemaHandler::getIndexes(std::string const& colName,
```

Comment:
I would suggest to hand in the LogicalCollection& here.
The caller has already done the CollectionNameResolver stunt to get this Pointer with error checking.
No need to do this here again.